src/common/src/util/sort_util.rs

```diff
@@ -36,81 +36,185 @@ pub enum Direction {
+        match direction {
+            PbDirection::Ascending => Self::Ascending,
+            PbDirection::Descending => Self::Descending,
             PbDirection::Unspecified => unreachable!(),
         }
     }
 
     pub fn to_protobuf(self) -> PbDirection {
         match self {
-            Direction::Ascending => PbDirection::Ascending,
-            Direction::Descending => PbDirection::Descending,
+            Self::Ascending => PbDirection::Ascending,
+            Self::Descending => PbDirection::Descending,
         }
     }
 }
 
-#[allow(dead_code)]
+/// Nulls are largest/smallest.
 #[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Display, Default)]
-pub enum NullsAre {
+enum NullsAre {
     #[default]
     #[display("LARGEST")]
     Largest,
     #[display("SMALLEST")]
     Smallest,
 }
 
+impl NullsAre {
+    pub fn from_protobuf(nulls_are: &PbNullsAre) -> Self {
+        match nulls_are {
+            PbNullsAre::Largest => Self::Largest,
+            PbNullsAre::Smallest => Self::Smallest,
+            PbNullsAre::Unspecified => unreachable!(),
+        }
+    }
+
+    pub fn to_protobuf(self) -> PbNullsAre {
+        match self {
+            Self::Largest => PbNullsAre::Largest,
+            Self::Smallest => PbNullsAre::Smallest,
+        }
+    }
+}
+
+/// Order type of a column.
 #[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Default)]
 pub struct OrderType {
     direction: Direction,
-    // TODO(rc): enable `NULLS FIRST | LAST`
-    // nulls_are: NullsAre,
+    nulls_are: NullsAre,
 }
 
 impl OrderType {
     pub fn from_protobuf(order_type: &PbOrderType) -> OrderType {
         OrderType {
             direction: Direction::from_protobuf(&order_type.direction()),
+            nulls_are: NullsAre::from_protobuf(&order_type.nulls_are()),
         }
     }
 
     pub fn to_protobuf(self) -> PbOrderType {
         PbOrderType {
             direction: self.direction.to_protobuf() as _,
+            nulls_are: self.nulls_are.to_protobuf() as _,
         }
     }
 }
 
 impl OrderType {
-    pub const fn new(direction: Direction) -> Self {
-        Self { direction }
+    fn new(direction: Direction, nulls_are: NullsAre) -> Self {
+        Self {
+            direction,
+            nulls_are,
+        }
+    }
+
+    fn nulls_first(direction: Direction) -> Self {
+        match direction {
+            Direction::Ascending => Self::new(direction, NullsAre::Smallest),
+            Direction::Descending => Self::new(direction, NullsAre::Largest),
+        }
+    }
+
+    fn nulls_last(direction: Direction) -> Self {
+        match direction {
+            Direction::Ascending => Self::new(direction, NullsAre::Largest),
+            Direction::Descending => Self::new(direction, NullsAre::Smallest),
+        }
     }
 
-    /// Create an ascending order type, with other options set to default.
-    pub const fn ascending() -> Self {
+    pub fn from_bools(asc: Option<bool>, nulls_first: Option<bool>) -> Self {
```

Comment:
Seems not a good interface design as it's hard to tell why some variant should be the `true` case. 😂 But if our frontend follows this convention, then it should be okay.

Reply:
You can think of this as the only pub interface used by frontend binder, and the semantics of these two parameters follow the convention in frontend.